src/realm/group.cpp

```diff
@@ -899,13 +901,13 @@ Table* Group::do_add_table(StringData name, bool is_embedded, bool do_repl)
 
     Replication* repl = *get_repl();
     if (do_repl && repl)
-        repl->add_class(key, name, is_embedded);
+        repl->add_class(key, name, table_type);
 
     ++m_num_tables;
 
     Table* table = create_table_accessor(j);
-    if (is_embedded)
-        table->do_set_embedded(true);
+    if (table_type == Table::Type::Embedded)
```

Comment:
You should unconditionally set the table type.

Reply:
Good point. Although, before is was only set if embedded.

Reply:
Yes, because that was the only case where you needed to set anything. Now there is both embedded and asymmetric.